torch/csrc/jit/codegen/cuda/runtime/memory.cu

```diff
@@ -0,0 +1,107 @@
+// Utility macro for this file
+#define DEVICE_INLINE __device__ inline
+
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 750))
+
+namespace Turing {
+
+namespace util {
+
+// Special utility for ldmatrix and cp_async
+DEVICE_INLINE unsigned toSmem(const void* ptr) {
+  unsigned smem_ptr;
+
+  asm("{ .reg .u64 smem_ptr; cvta.to.shared.u64 smem_ptr, %1; cvt.u32.u64 %0, smem_ptr; }\n"
+      : "=r"(smem_ptr)
+      : "l"(ptr));
+
+  return smem_ptr;
+}
+
+} // namespace util
+
+DEVICE_INLINE void ldMatrix(Array<__half, 4, 4>& out, void const* ptr) {
+  uint2& val = reinterpret_cast<uint2&>(out);
+  unsigned addr = util::toSmem(ptr);
+  asm volatile("ldmatrix.sync.aligned.x2.m8n8.shared.b16 {%0,%1}, [%2];"
+               : "=r"(val.x), "=r"(val.y)
+               : "r"(addr));
+}
+
+DEVICE_INLINE void ldMatrixT(Array<__half, 4, 4>& out, void const* ptr) {
```

Comment:
Same as previous, 8x8 matrix is vectorized loaded, then scattered (to perform transpose) so threads will hold 2 values down a column (instead of the previous instruction that's across a row).

Reply:
Added to comment. Thanks.